File: Software Development Module/Revision/Exam2015/SchoolSuppliesApp.java
/*
SchoolSuppliesApp.java
Billy Cussen
10/08/2020
*/

import java.util.Scanner;
import java.util.InputMismatchException;

public class SchoolSuppliesApp{

    public static void main(String [] args){

        //Variable
        int supplyTypes = 0;
        int[] supplyQuantity;
        double[] supplyCost;
        String total;
        boolean inputCheck;

        //Objects
        Scanner keyboard;
        SchoolSupplies mySchoolSupplies;

        //Input
        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);
            
            try{
                System.out.print("Please enter the number of Supply Types you will be requesting: ");
                supplyTypes = keyboard.nextInt();

                if(supplyTypes <= 0){
                    throw new InputMismatchException();
                }
            }catch(InputMismatchException | NumberFormatException e){
                inputCheck = false;
                System.out.println("You need to enter a number greater than 0");
            }finally{
                keyboard = null;
            }

        }while(!inputCheck);

        supplyQuantity = new int[supplyTypes];
        supplyCost = new double[supplyTypes];

        do{

            for(int i = 0; i < supplyQuantity.length; i++){

                inputCheck = true;
                keyboard = new Scanner(System.in);

                try{
                    
                    System.out.print("Please enter the quantity of Supply type "+(i+1)+" that you require: ");
                    supplyQuantity[i] = keyboard.nextInt();

                    if(supplyQuantity[i] <= 0){
                        throw new InputMismatchException();
                    }
                }catch(InputMismatchException | NumberFormatException e){
                    inputCheck = false;
                    System.out.println("You need to enter a number greater than 0");
                }finally{
                    keyboard = null;
                }

            }

        }while(!inputCheck);

        do{

            for(int i = 0; i < supplyCost.length; i++){

                inputCheck = true;
                keyboard = new Scanner(System.in);

                try{
                    
                    System.out.print("Please enter the Cost of Supply type "+(i+1)+": ");
                    supplyCost[i] = keyboard.nextDouble();

                    if(supplyCost[i] <= 0){
                        throw new InputMismatchException();
                    }
                }catch(InputMismatchException | NumberFormatException e){
                    inputCheck = false;
                    System.out.println("You need to enter a number greater than 0");
                }finally{
                    keyboard = null;
                }

            }

        }while(!inputCheck);

        //Process
        mySchoolSupplies = new SchoolSupplies();
        total = mySchoolSupplies.calculateSchoolSupplies(supplyQuantity,supplyCost);
        mySchoolSupplies = null;
        System.out.println(total);
    }

}

File: Python Course/Software Development Module - in Python/Revision/Exam2015/SchoolSuppliesApp.py
"""
SchoolSuppliesApp.py
Billy Cussen
28/10/2020
"""

#Object
import gc
import Encoder
from SchoolSupplies import mySchoolSupplies

repeat = "y"

while repeat == "y":

    inputCheck = False
    #Input - Supply Types?
    while inputCheck == False:
        try: 
            quantity = int(input("Please enter the Number of Supply Types you will be requesting: "))
            if quantity <= 0:
                raise Exception
        except Exception as e:
            print("You need to enter a number greater than 0")
        else:
            inputCheck = True

    supplyQuantity = []
    supplyCost = []
    iterator = 0

    #Quantity of each!
    while iterator < quantity:
        inputCheck = True
        while inputCheck:
            try: 
                numTemp = int(input("Please enter the Quantity of Supply Type "+str(iterator+1)+": "))
                if numTemp <= 0:
                    raise Exception
            except Exception as e:
                print("You need to enter a number greater than 0")
            else:
                inputCheck = False
                supplyQuantity.append(numTemp)

        iterator=iterator+1
    
    #Cost of each!
    iterator = 0
    while iterator < quantity:
        inputCheck = True
        while inputCheck:
            try: 
                numTemp = float(input("Please enter the Cost of Supply Type "+str(iterator+1)+": "))
                if numTemp <= 0:
                    raise Exception

            except Exception as e:
                print("You need to enter a number greater than 0")
            else:
                inputCheck = False
                supplyCost.append(numTemp)
        iterator=iterator+1

    #Process and Output
    mySchoolSuppliesApp = mySchoolSupplies(supplyQuantity, supplyCost)
    print(mySchoolSuppliesApp.calculateSupplies())
    mySchoolSuppliesApp = None
    gc.collect()

    #Input - Repeat?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter \"y\" to retry, \"n\" to exit: "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print("You need to enter \"y\" or \"n\"")

print("Thanks for using my School Supply App!")